### ISSUE TITLE
Support of `theta_init` of `Matrix` type / GPU-level multiple chain support

### ISSUE THREAD

**Issue by xukai92** [TuringLang/AdvancedHMC.jl#92]
Current `theta_init` is assumed to be a vecotor to run a single chain of MCMC.
It is possible to allow `theta_init` be a matrix where each column is an initial point,
and run many chains in parallel by allowing all internal computation in a "batch manner".
I'm not sure how it would speed things in CPUs but it definitely would accerlate things on GPUs.

**Comment by xukai92** [comment]
Another issue is how would a dynamic trajectory work in this case.

**Comment by xukai92** [comment]
Closed by https://github.com/TuringLang/AdvancedHMC.jl/pull/117
